### PR TITLE
net-p2p/go-ipfs-bin: Fix systemd units

### PR DESCRIPTION
**Pull request by kakra**

### CODE REVIEW

**Comment by juippis** [comment]
Would you like to update to GLEP-81 method of handling users/groups in ebuilds? Reserve UID/GID in gentoo-dev mailing list, then add two simple ebuilds to this PR. 
You'll need to do it at some point, but I understand if you prefer to leave it later since this is a simple PR.

**Comment by kakra** [comment]
Yes I can do that.

**Comment by stkw0** [comment]
@kakra are you interested in co-maintain go-ipfs{,-bin}? Currently I don't have too much time.

**Comment by kakra** [comment]
@stkw0 Yes I can do that. Should I add myself to the maintainer list for the next PR? I already maintain bees and gamemode.

**Comment by kakra** [comment]
I've prepared everything in my local branch and sent a uid/gid assignment RFC to gentoo-dev.

**Comment by juippis** [comment]
If it doesn't receive comments, its good. Could you still push the acct-*/go-ipfs-bin (or just plain ipfs?) packages in this PR, and update go-ipfs-bin ebuild to use them?

**Comment by kakra** [comment]
> If it doesn't receive comments, its good.

@juippis That wasn't clear to me from the GLEP-81 text.

@stkw0 I'm not sure about when and why to remove older ipfs versions. I think it should be fine to drop just any older version.

**Comment by stkw0** [comment]
> > If it doesn't receive comments, its good.
> 
> @juippis That wasn't clear to me from the GLEP-81 text.
> 
> @stkw0 I'm not sure about when and why to remove older ipfs versions. I think it should be fine to drop just any older version.

I usually try to leave two older versions just in case someone have problems with the new one. In any case, at some point in near future we should abandon go-ipfs-bin in favor of go-ipfs (see PR #12412 ).
I will copy your systemd files and add you as a maintainer in the new go-ipfs package as I understand you are also interested in it.

Thank you very much.

**Comment by kakra** [comment]
> I will copy your systemd files

@stkw0 I've amended the commits to make both systemd service files act according to your documentation of setting `IPFS_PATH` in `/etc/conf.d`. The instantiated version (with `@`) should act like a multi-instance openrc service does - tho in my mind it doesn't make sense to run more than one instance on the same system, and I couldn't test this (I no longer run any openrc systems).

File: net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild [hunk 1]
```diff
@@ -17,7 +17,12 @@ LICENSE="MIT"
 SLOT="0"
 KEYWORDS="-* ~amd64 ~arm ~x86"
 
-RDEPEND="sys-fs/fuse"
+RDEPEND="
+	acct-group/syncthing
+	acct-user/syncthing
```
Comment (stkw0) [inline net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild]: Small error here, it should be ipfs not syncthing :p

Comment (kakra) [reply net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild]: Oh noes! Now you know my secret... ;-)

Okay, fixed.

**Comment by stkw0** [comment]
> > I will copy your systemd files
> 
> @stkw0 I've amended the commits to make both systemd service files act according to your documentation of setting `IPFS_PATH` in `/etc/conf.d`. The instantiated version (with `@`) should act like a multi-instance openrc service does - tho in my mind it doesn't make sense to run more than one instance on the same system, and I couldn't test this (I no longer run any openrc systems).

I never used a multi-instance service, the IPFS_PATH setting was just in case someone needed to specify a different location for some reason. Maybe I misunderstood something ¿?

**Comment by kakra** [comment]
@juippis Gentle ping? I think it should be ready for merge... Please have a look.

File: net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit bash-completion-r1 systemd user
```
Comment (juippis) [inline net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild]: You can drop user.eclass

File: net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	https://raw.githubusercontent.com/ipfs/go-ipfs/v${PV}/misc/completion/ipfs-completion.bash -> ${P}.bash"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~arm ~x86"
+
+RDEPEND="
+	acct-group/ipfs
+	acct-user/ipfs
+	sys-fs/fuse
```
Comment (juippis) [inline net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild]: Needs to be slotted, probably `:0` for now (but if you want, you can test if it works with fuse-3.6 too, then `:*` for slot).

Rest looks good, sorry for not following up!

Comment (kakra) [reply net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild]: Let's stick to `:0` for now and check compatibility with other slots later when we add the non-bin ebuild variant.

File: net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild (hunk 2 above)
Comment (kakra) [reply net-p2p/go-ipfs-bin/go-ipfs-bin-0.4.22.ebuild]: Okay, removed.

**Comment by juippis** [comment]
Thanks!

**Pull request closed without merging**